src/libstore/store-api.cc

```diff
@@ -1442,6 +1443,10 @@ std::shared_ptr<Store> openFromNonUri(const std::string & uri, const Store::Para
         return std::make_shared<UDSRemoteStore>(params);
     } else if (uri == "local") {
         return std::make_shared<LocalStore>(params);
+    } else if (uri == "local-overlay") {
```

Comment:
If we use `local-overlay://` we do not need this. But we don't yet have something to go after the `://`. (Same issue with `local`.)